util/route.libsonnet

```diff
@@ -17,6 +17,31 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
       + route.spec.to.withName(service.metadata.name)
       + route.spec.port.withTargetPort(port),
 
+    '#weightedRouteFor': d.fn(|||
+      weightedRouteFor constructs a openshift route to the specified services. It expects
+      a value for the name, hostname, port, services and defaults to '/' for the path.
+      services expects an array consisting of objects with the keys name (name of the service) and weight.
+    |||, [
+      d.arg('name', d.T.string),
+      d.arg('host', d.T.string, ''),
+      d.arg('port', d.T.number, ''),
+      d.arg('services', d.T.object),
+      d.arg('path', d.T.string, '/'),
+    ]),
+    weightedRouteFor(name, host, port, services, path='/')::
+      route.new(name, host, path)
+      + route.spec.to.withKind('Service')
+      + route.spec.to.withName(services[0].name)
+      + route.spec.to.withWeight(services[0].weight)
+      + route.spec.port.withTargetPort(port)
+      + route.spec.withAlternateBackends(
+        std.prune(
```

Comment:
I think using slice syntax here might be more readable (had to look up the documentation for `prune` to understand what exactly is happening here)

For example `services[1:std.length(services):1]`